hathor/p2p/resources/add_peers.py

```diff
@@ -56,8 +56,8 @@ def already_connected(connection_string: str) -> bool:
             # determines if given connection string is already among connected or connecting peers
             endpoint_url = connection_string.replace('//', '')
 
-            # remove peers that we're already trying to connect
-            if endpoint_url in self.manager.connections.connecting_peers:
+            # ignore peers that we're already trying to connect
+            if endpoint_url in set(self.manager.connections.iter_not_ready_endpoints()):
```

Comment:
I guess we do not need this `set(...)` here.

Reply:
Removed.